src/main.cpp

```diff
@@ -1,6 +1,6 @@
 #include <string>
 #include "world.hpp"
-#include "lua.hpp"
+#include "lua/lua.hpp"
```

Comment:
no; lua.hpp is inside src and we intend to include that one

Reply:
lua.hpp in src folder shadows <lua.hpp>, which makes it impossible to build. Either they should have different folder or lua in src folder should be renamed.

Reply:
It only shadows the one in include if you use ```-I```, that is why we moved to using ```-iquote``` some time ago.

Reply:
When it is "" include, it searches 1st in relative path, and when it is <> it searches first in system path. However, it seems in windows it failed to find correct one as both the original lua path and the our lua path are as system includes.

Reply:
>Directories specified with -iquote apply only to the quote form of the directive, #include "file". Directories specified with -I, -isystem, or -idirafter apply to lookup for both the #include "file" and #include <file> directives.
https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html#Directory-Options

It should not even be applicaple for the one which searches system path

Reply:
windows is the problematic case sadly. There you can use both, just the ordering is different.

Reply:
What compiler were you using?

Reply:
vc++ one, I don't have mingw installed. vc problem is also not a new.

Reply:
> lua.hpp in src folder shadows <lua.hpp>, which makes it impossible to build. Either they should have different folder or lua in src folder should be renamed.

Can confirm, this was a problem on Windows, so I just renamed our lua.hpp to lua.h in my PR

Reply:
There is lua.h header as well. lua.hpp is basicaly just a extern "C" { #include "lua.h"; } and 2 more headers. Best choice would be unique name so that it is sure not to have name collision.

Reply:
>vc++ one, I don't have mingw installed. vc problem is also not a new.

Yeah msvc doesn't have anything like ```-iquote```.
Although we could use some header magic with ```/FI``` and ```#define _WIN32``` but that is rather dirty

Reply:
or just change the name, honestly it is not requirement for them to shadow each other.

Reply:
Yes.
I should've mentioned that I prefer just renaming it.